.github/scripts/auto_release.py

```
@@ -0,0 +1,274 @@
+    new_version: str = Field(
+        description="The new version number in X.Y.Z format"
+    )
+    changelog_entry: str = Field(
+        description="Full changelog entry in markdown format with ## [X.Y.Z] header"
+    )
+
+
+def run_command(cmd: str, check: bool = True) -> str:
+    """Run a shell command and return output."""
+    print(f"Running: {cmd}")
+    result = subprocess.run(
```

Ruff (0.14.1)

`subprocess` call with `shell=True` identified, security issue

(S602)